homeassistant/components/plex/config_flow.py

```diff
@@ -0,0 +1,355 @@
+        """
+        if self._async_in_progress():
+            return self.async_abort(reason="already_in_progress")
+
+        host_and_port, host_config = import_config.popitem()
+        prefix = "https" if host_config[CONF_SSL] else "http"
+        url = "{}://{}".format(prefix, host_and_port)
+
+        config = {
+            CONF_URL: url,
+            CONF_TOKEN: host_config[CONF_TOKEN],
+            CONF_VERIFY_SSL: host_config["verify"],
+        }
+
+        _LOGGER.info("Imported configuration from legacy config file")
+        return await self.async_step_user(user_input=config)
+
+    async def async_step_import_sensor(self, import_config):
+        """Import from Plex Sensor configuration.
+
+        Legacy.
+        """
+        if self._async_in_progress():
+            return self.async_abort(reason="already_in_progress")
+
+        host = import_config.get(CONF_HOST)
+        port = import_config.get(CONF_PORT)
+        username = import_config.get(CONF_USERNAME)
+        password = import_config.get(CONF_PASSWORD)
+        token = import_config.get(CONF_TOKEN)
+        server = import_config.get(CONF_SERVER)
+
+        if username:
+            if password:
+                _LOGGER.warning(
+                    "Password no longer supported, please set up via Integrations"
+                )
+                return self.async_abort(reason="password_provided")
+
+            if token:
+                config = {
+                    CONF_USERNAME: username,
+                    CONF_TOKEN: token,
+                    CONF_SERVER: server,
+                }
+        else:
+            prefix = "https" if import_config[CONF_SSL] else "http"
+            url = "{}://{}:{}".format(prefix, host, port)
+
+            config = {
+                CONF_URL: url,
+                CONF_TOKEN: token,
+                CONF_VERIFY_SSL: import_config[CONF_VERIFY_SSL],
+            }
+
+        _LOGGER.info("Imported Plex credentials from sensor configuration")
+        return await self.async_step_user(user_input=config)
+
+
+class PlexOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Plex options."""
+
+    def __init__(self, config_entry):
+        """Initialize Plex options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+
+    async def async_step_init(self, user_input=None):
+        """Manage the Plex options."""
+        return await self.async_step_plex_platforms()
+
+    async def async_step_plex_platforms(self, user_input=None):
+        """Choose Plex platforms to enable."""
+        if user_input is not None:
+            self.options[CONF_ENABLE_MEDIA_PLAYER] = user_input[
+                CONF_ENABLE_MEDIA_PLAYER
+            ]
+            self.options[CONF_ENABLE_SENSOR] = user_input[CONF_ENABLE_SENSOR]
+            if user_input[CONF_ENABLE_MEDIA_PLAYER]:
+                return self.async_show_form(
+                    step_id="plex_mp_settings",
```

Comment:
This should be done in the Plex mp settings method

Reply:
Good call, it felt a little off.